src/formatDate.js

```diff
@@ -50,7 +50,55 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatSeparator = fromFormat[3];
+  const toFormatSeparator = toFormat[3];
+
+  const fromYearFormat = fromFormat.find(item => item.includes('Y'));
+  const toYearFormat = toFormat.find(item => item.includes('Y'));
+
+  const dateToFormat = date.split(fromFormatSeparator);
+
+  const yearIndex = fromFormat.indexOf(fromYearFormat);
+  const monthIndex = fromFormat.indexOf('MM');
+  const dayIndex = fromFormat.indexOf('DD');
+
+  let dateYear = dateToFormat[yearIndex];
+  const dateMonth = dateToFormat[monthIndex];
+  const dateDay = dateToFormat[dayIndex];
+
+  const dateYearShortened = +(dateYear.slice(0, 2));
+
+  const result = [];
+
+  if (fromYearFormat.length > toYearFormat.length) {
+    dateYear = dateYear.slice(2);
+  }
+
+  if (fromYearFormat.length < toYearFormat.length
```

Comment:
else if?
you no need to check the second condition if the first was true (and the same further)

Reply:
I think I need, because there is the third condition: fromYearFormat.length === toYearFormat.length, and I should do nothing if it is true. So i can't just check only one condition. The only way to avoid repeating the same code further is nesting one if inside another